src/formatMessagesAsString/index.js

```diff
@@ -0,0 +1,32 @@
+import chalk from 'chalk'
+
+/**
+ * @param {Array<Object>} messages: An array of message objects.
+ * @returns {String} A colored string representation of the given message objects.
+ */
+export default function formatMessagesAsString(messages) {
+  // Return undefined when there are no messages to format
+  if (!messages.length) {
+    return undefined
+  }
+
+  return messages.reduce(messageReducer, '')
+
+  function messageReducer(accumulatorString, messageObject, index, array) {
+    // Don't append newline to the last entry
+    const newline = index === array.length - 1 ? '\n' : ''
+
+    const formattedMessage = formatMessage(messageObject)
+    return `${accumulatorString}${newline}${formattedMessage}\n`
+  }
+}
```

Comment:
I started using `reduce` for things a few months ago and I totally love it. It's crazy powerful [example from yesterday](https://github.com/kentcdodds/webpack-validator/blob/65111bb9c80fd5e9771ad9ec14bf02dd116275c5/src/validators/stats/index.js#L24-L29). I found [these lessons](https://egghead.io/series/reduce-data-with-javascript) are reeeally good.